test-basic/documents-core.js

```diff
@@ -57,6 +57,40 @@ describe('document content', function(){
             })
           .catch(done);
       });
+        it('should read back zipped contents with enableGzippedResponses as true', function(done){
+            testconfig.restReaderConnection.enableGzippedResponses = true;
+            const dbReader = marklogic.createDatabaseClient(testconfig.restReaderConnection);
+            dbReader.documents.read('/test/write/string1.json')
+                .result(function(documents) {
+                    valcheck.isUndefined(documents).should.equal(false);
+                    documents.length.should.equal(1);
+                    let document = documents[0];
+                    document.should.have.property('content');
+                    try{
+                        document.content.should.have.property('key1');
+                    } catch(error){
+                        // An AssertionError should be thrown here since the contents are zipped and cannot have property key1.
+                        done();
+                    }
+                })
```

Comment:
Verified in responder by logging - response.headers -
{
  'vnd.marklogic.document-format': 'json',
  'content-type': 'application/json; charset=utf-8',
  'ml-effective-timestamp': '17013030096192060',
  server: 'MarkLogic',
  'content-encoding': 'gzip',
  connection: 'close'
}